src/device.rs

```diff
@@ -2285,7 +2290,10 @@ impl TaskRuntime for Runtime {
                                 direct_entities.cross_ping_check.notify_failed_wg_connection(public_key)
                                     .await?;
 
-                                direct_entities.session_keeper.remove_node(&public_key).await?;
+
+                                if self.features.batching.is_some() {
+                                    direct_entities.session_keeper.remove_node(&public_key).await?;
+                                };
```

Comment:
I'm not sure why this is done conditionally. If that is the correct behaviour here it would be good to explain that in comment.

Reply:
My bad. I was under impression that original implementation never did delete the entries so wanted to leave the behaviour the same. Now it's apparent that it's wrong. Thanks for picking it up, I've patched it